src/app/enrolled-course-details-page/enrolled-course-details-page.ts

```diff
@@ -1761,6 +1761,7 @@ export class EnrolledCourseDetailsPage implements OnInit, OnDestroy {
         this.objRollup,
         this.corRelationList
       );
+      this.resetSavedQuizContent();
```

Comment:
It only solves the enrolling of guest users what if any other scenario comes up. Can we put that logic where we are navigating to the Unlisted ContentDetails page which will solve all other scenarios.